CHANGELOG.md

```diff
@@ -3,16 +3,21 @@
 All notable changes to this project will be documented in this file.
 
 ## Unreleased
-
-### Forthcoming changes
-This release deprecates the `buildkite_organization_settings` [resource](./docs/resources/organization_settings.md). In a future minor release, we will remove this resource in favour of the newer `buildkite_organization` [resource](./docs/resources/organization.md) that aligns with the [datasource](./docs/data-sources/organization.md) of the same name. 
-
 * Migrate organization datasource to plugin framework [[PR #304](https://github.com/buildkite/terraform-provider-buildkite/pull/304)] @jradtilbrook
 * SUP-1051: Team Member resource Framework/Genqlient conversion [[PR #313](https://github.com/buildkite/terraform-provider-buildkite/pull/313)] @james2791
 * SUP-1067: Organization resource conversion to Framework[[PR #311](https://github.com/buildkite/terraform-provider-buildkite/pull/311)] @james2791
 * SUP-1063 Convert meta datasource to framework [[PR #314](https://github.com/buildkite/terraform-provider-buildkite/pull/314)] @jradtilbrook
 * SUP-1065: Pipeline Datasource conversion to Framework [[PR #315](https://github.com/buildkite/terraform-provider-buildkite/pull/315)] @james2791
 * Convert DeletePipeline to genqlient [[PR #317](https://github.com/buildkite/terraform-provider-buildkite/pull/317)] @jradtilbrook
+* SUP-1049 Migrate Team to framework & Genqlient [[PR #318](https://github.com/buildkite/terraform-provider-buildkite/pull/317)] @mcncl
+
+### Forthcoming changes
+This release deprecates the `buildkite_organization_settings` [resource](./docs/resources/organization_settings.md). In a future minor release, we will remove this resource in favour of the newer `buildkite_organization` [resource](./docs/resources/organization.md) that aligns with the [datasource](./docs/data-sources/organization.md) of the same name. 
+[[PR #318](https://github.com/buildkite/terraform-provider-buildkite/pull/317)] introduces a change in how we handle
+
+Team data-sources importing Teams. Instead of `slug` the data-source will now require ID as the input. This is to avoid
+breaking changes which *may* be caused with the rename of a Team and the resulting change of `slug` value. GraphQL IDs
+of resources to not change, regardless of values within the resource being changed. 
```

Comment:
Ohh I misunderstood our chat. This isn't related to a `terraform import`, its related to how to identify the team for loading through a datasource.

In that case, I think it would be better not to use the word "importing" in this description because of the possibility of confusion there.  
I also think this will be a breaking change in the next release for anyone already using the datasource through slug -- and in that case, I'd like to support a deprecation/migration period and then switch over.

I actually think some other providers (eg github, aws) allow their datasources to use more than one identifier to load the resource and implement the searching functionality themselves. That way, users can use a human readable thing like slug to load it and then reference the ID of the datasource making a nicer UX. So we could perhaps support both and document the caveat when using slug?